core/src/test/java/org/elasticsearch/action/support/AutoCreateIndexTests.java

```diff
@@ -121,27 +130,59 @@ public void testAutoCreationMultiplePatternsNoWildcards() {
-        AutoCreateIndex autoCreateIndex = new AutoCreateIndex(settings, new IndexNameExpressionResolver(settings));
+        Settings settings = Settings.builder().put(AutoCreateIndex.AUTO_CREATE_INDEX_SETTING.getKey(),
+                "+test1,-test1,-test2,+test2").build();
+        AutoCreateIndex autoCreateIndex = createAutoCreateIndex(settings);
         ClusterState clusterState = ClusterState.builder(new ClusterName("test")).metaData(MetaData.builder()).build();
         assertThat(autoCreateIndex.shouldAutoCreate("test1", clusterState), equalTo(true));
         assertThat(autoCreateIndex.shouldAutoCreate("test2", clusterState), equalTo(false));
         assertThat(autoCreateIndex.shouldAutoCreate("does_not_match" + randomAsciiOfLengthBetween(1, 5), clusterState), equalTo(false));
     }
 
+    public void testUpdate() {
+        boolean value = randomBoolean();
+        Settings settings;
+        if (value && randomBoolean()) {
+            settings = Settings.EMPTY;
+        } else {
+            settings = Settings.builder().put(AutoCreateIndex.AUTO_CREATE_INDEX_SETTING.getKey(), value).build();
+        }
+
+        ClusterSettings clusterSettings = new ClusterSettings(settings,
+                ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+        AutoCreateIndex  autoCreateIndex = new AutoCreateIndex(settings, clusterSettings, new IndexNameExpressionResolver(settings));
+        assertThat(autoCreateIndex.getAutoCreate().isAutoCreateIndex(), equalTo(value));
+
+        Settings newSettings = Settings.builder().put(AutoCreateIndex.AUTO_CREATE_INDEX_SETTING.getKey(), !value).build();
+        clusterSettings.applySettings(newSettings);
+        assertThat(autoCreateIndex.getAutoCreate().isAutoCreateIndex(), equalTo(!value));
+
+        newSettings = Settings.builder().put(AutoCreateIndex.AUTO_CREATE_INDEX_SETTING.getKey(), "logs-*").build();
+        clusterSettings.applySettings(newSettings);
+        assertThat(autoCreateIndex.getAutoCreate().isAutoCreateIndex(), equalTo(true));
+        assertThat(autoCreateIndex.getAutoCreate().getExpressions().size(), equalTo(1));
+        assertThat(autoCreateIndex.getAutoCreate().getExpressions().get(0).v1(), equalTo("logs-*"));
+    }
+
     private static ClusterState buildClusterState(String... indices) {
         MetaData.Builder metaData = MetaData.builder();
         for (String index : indices) {
             metaData.put(IndexMetaData.builder(index).settings(settings(Version.CURRENT)).numberOfShards(1).numberOfReplicas(1));
         }
-        return ClusterState.builder(org.elasticsearch.cluster.ClusterName.CLUSTER_NAME_SETTING.getDefault(Settings.EMPTY)).metaData(metaData).build();
+        return ClusterState.builder(org.elasticsearch.cluster.ClusterName.CLUSTER_NAME_SETTING.getDefault(Settings.EMPTY))
+                .metaData(metaData).build();
+    }
+
+    private AutoCreateIndex createAutoCreateIndex(Settings settings) {
```

Comment:
the double 'create' in the method name confused me a bit when reading it, maybe rename to 'newAutoCreateIndex'?